src/game/movement.rs

```diff
@@ -25,9 +32,30 @@ pub(super) fn plugin(app: &mut App) {
     );
 }
 
-#[derive(Component, Reflect, Default)]
+/// These are the movement parameters for our character controller.
+/// For now, this is only used for a single player, but it could power NPCs or other players as well.
+#[derive(Component, Reflect)]
 #[reflect(Component)]
-pub struct MovementController(pub Vec2);
+pub struct MovementController {
+    /// The direction the character wants to move in.
+    pub intent: Vec2,
+
+    /// Since Bevy's default 2D camera setup is scaled such that
+    /// one unit is one pixel, you can think of this as
+    /// "How many pixels per second should the player move?"
+    /// Note that physics engines may use different unit/pixel ratios.
+    pub max_speed: f32,
+}
+
+impl Default for MovementController {
+    fn default() -> Self {
+        Self {
+            intent: Vec2::ZERO,
+            // 400 pixels per second is a nice default, but we can still vary this per character.
+            max_speed: 400.0,
+        }
+    }
+}
 
 fn record_movement_controller(
```

Comment:
I'm kinda feeling like we should rename this system. How about `record_directional_input` or `record_keyboard_movement_input`? Since the input is what we're recording, and we're recording it _into_ the movement controller. But there may be other systems that can record other things into the movement controller.

Reply:
[`929a7a1` (#230)](https://github.com/TheBevyFlock/bevy_quickstart/pull/230/commits/929a7a1886a0fa8bea889bdaaed6ea0bd627375d)

Reply:
If I'm being super pedantic, it's not tied to the player. But since it's using keyboard input, which is _probably_ meant to control the player entity, it basically is. So it's fine.